### ISSUE TITLE
Error with Arrays.asList

### ISSUE THREAD

**Issue by jwaataja** [t-rasmud/checker-framework#93]
Consider the following class
```java
import org.checkerframework.checker.determinism.qual.*;
import java.util.*;

public class AsListTest {
    public static void f() {
        @Det List<@Det String> list = Arrays.<@Det String>asList("");
    }
}
```
```
AsListTest.java:6: error: [invalid.element.type] element type (@Det) cannot be used with collection type (@NonDet)
        @Det List<@Det String> list = Arrays.<@Det String>asList("");
                                                                ^
AsListTest.java:6: error: [type.argument.type.incompatible] incompatible types in type argument.
        @Det List<@Det String> list = Arrays.<@Det String>asList("");
                                              ^
  found   : @Det String
  required: [extends @NonDet Object super @NonDet null]
```

The first error is incorrect because the `asList` method should be returning a `@Det List` not `@NonDet`. The second error is wrong because the method should be able to accept `@Det` type parameters. This would probably be fixed by updating the JDK annotations.

**Comment by mernst** [comment]
@t-rasmud Did you fix this?

**Comment by smillst** [comment]
These can be fixed in Randoop by removing the explicit type argument:
```
new TypeTuple(Arrays.asList(JavaTypes.OBJECT_TYPE, JavaTypes.OBJECT_TYPE))
```
When Randoop was compiled with Java 7, the explicit type argument was required.

**Comment by mernst** [comment]
I made a pull request: https://github.com/jwaataja/randoop/pull/5

**Comment by smillst** [comment]
FYI this only works because Java 8 type inference isn’t implemented and most warning are automatically suppressed. So Rashmi’s changes are still important.

**Comment by mernst** [comment]
@smillst Thanks for that clarification.

**Comment by t-rasmud** [comment]
Fixed in b05d84079790d25e5cbfad0355bb74484223bfb8
